### PR TITLE
[libzip] Update to version 1.8.0

### PR DESCRIPTION
**Pull request by ctabin**

### CODE REVIEW

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]
@ctabin ,Please add 'FEATURES' in the function"vcpkg_check_features" like:
````
vcpkg_check_features(
OUT_FEATURE_OPTIONS FEATURE_OPTIONS
FEATURES
    bzip2 ENABLE_BZIP2   
    liblzma ENABLE_LZMA    
    openssl ENABLE_OPENSSL   
    wincrypto ENABLE_WINDOWS_CRYPTO    
    commoncrypto ENABLE_COMMONCRYPTO   
    mbedtls ENABLE_MBEDTLS)

````

File: ports/libzip/vcpkg.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "libzip",
-  "version-semver": "1.7.3",
-  "port-version": 2,
+  "version-semver": "1.8.0",
+  "port-version": 0,
```
Comment (JonLiu1993) [inline ports/libzip/vcpkg.json:4]:
```suggestion
```

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

**Comment by JonLiu1993** [comment]
@ctabin ,Have you test all features locally in the following triplet:
- x64-windows
- x64-windows-static
- x64-linux

File: ports/libzip/portfile.cmake [hunk 2]
```diff
@@ -1,9 +1,9 @@
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO nih-at/libzip
-    REF 66e496489bdae81bfda8b0088172871d8fda0032 #v1.7.3
-    SHA512 ae0cda3e9decf5a71bf1e0907a2a21b2c0d83e6e576faf4d9401d6954707ae298c1c09febbc5339f457ace3577fdd405a790c819ef24778990ca6bf1e9516d54
-    HEAD_REF v1.7.3
+    REF 26ba5523db09213f532821875542dba7afa04b65 #v1.8.0
+    SHA512 caa4610e10a45260d8f06e4e728b231f0fcfacd90d3091a096b273997b7505857a78a8013d0571c3b25543d894eb049d1e7f5385e910066e464b3d208390570f
+    HEAD_REF v1.8.0
```
Comment (Hoikas) [inline ports/libzip/portfile.cmake:6]:
```suggestion
    HEAD_REF master
```

This should be the ref used to build the latest head, not the current release.

File: ports/libzip/vcpkg.json (hunk 1 above)
Comment (ctabin) [reply ports/libzip/vcpkg.json]: Removed.

File: ports/libzip/portfile.cmake (hunk 2 above)
Comment (ctabin) [reply ports/libzip/portfile.cmake]: Changed.

Comment (JonLiu1993) [reply ports/libzip/portfile.cmake]: > Changed.

Why not master?

File: ports/libzip/portfile.cmake [hunk 3]
```diff
@@ -1,21 +1,22 @@
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO nih-at/libzip
-    REF 66e496489bdae81bfda8b0088172871d8fda0032 #v1.7.3
-    SHA512 ae0cda3e9decf5a71bf1e0907a2a21b2c0d83e6e576faf4d9401d6954707ae298c1c09febbc5339f457ace3577fdd405a790c819ef24778990ca6bf1e9516d54
-    HEAD_REF v1.7.3
+    REF 26ba5523db09213f532821875542dba7afa04b65 #v1.8.0
+    SHA512 caa4610e10a45260d8f06e4e728b231f0fcfacd90d3091a096b273997b7505857a78a8013d0571c3b25543d894eb049d1e7f5385e910066e464b3d208390570f
+    HEAD_REF header
```
Comment (Hoikas) [inline ports/libzip/portfile.cmake:6]:
```suggestion
    HEAD_REF master
```

Typo?

**Comment by PhoebeHui** [comment]
@ctabin, can you address the review suggestions and the CI test failures?

**Comment by ctabin** [comment]
@PhoebeHui Sorry for the late reply. I'll make the changes asap !

**Review by JonLiu1993 — APPROVED** [review]

**Comment by JonLiu1993** [comment]
@ctabin ,Yesterday CI reported an error that the patch could not be applied. I re-made the patch and submitted it, but today it shows that there is still a problem under Linux. Could you please take a look?

**Comment by ctabin** [comment]
Hi @JonLiu1993,
Thanks for help on this. Unfortunately, since I'm not the maintainer/developer of libzip, it is quite hard for me to know exactly what's going on.
So far, in the CI logs, I see:

```
2021-07-28T13:34:07.4376012Z Error: Building package ignition-fuel-tools4:x64-linux failed with: BUILD_FAILED
2021-07-28T13:34:07.4376816Z Elapsed time for package ignition-fuel-tools4:x64-linux: 2.381 s
2021-07-28T13:34:07.4377495Z Starting package 152/152: libzippp:x64-linux
2021-07-28T13:34:07.4378143Z Building package libzippp[core]:x64-linux...
2021-07-28T13:34:10.0290862Z -- Downloading https://github.com/ctabin/libzippp/archive/8299422194ce3c5be0677550ce3d6d4e15d40dd8.tar.gz -> ctabin-libzippp-8299422194ce3c5be0677550ce3d6d4e15d40dd8.tar.gz...
2021-07-28T13:34:10.0292386Z -- Extracting source /mnt/vcpkg-ci/downloads/ctabin-libzippp-8299422194ce3c5be0677550ce3d6d4e15d40dd8.tar.gz
2021-07-28T13:34:10.0293151Z -- Applying patch fix-find-lzma.patch
2021-07-28T13:34:10.0293894Z -- Using source at /mnt/vcpkg-ci/buildtrees/libzippp/src/4e15d40dd8-75ef2fee11.clean
2021-07-28T13:34:10.0294532Z -- Configuring x64-linux-dbg
2021-07-28T13:34:10.0295095Z -- Configuring x64-linux-rel
2021-07-28T13:34:10.0295860Z -- Building x64-linux-dbg
2021-07-28T13:34:10.0296382Z -- Building x64-linux-rel
2021-07-28T13:34:10.0297067Z -- Installing: /mnt/vcpkg-ci/packages/libzippp_x64-linux/share/libzippp/copyright
2021-07-28T13:34:10.0297719Z -- Performing post-build validation
2021-07-28T13:34:10.0298312Z -- Performing post-build validation done
```
So it looks like it fails because with the current changes, `libzippp` (which is my library, that I updated in #18553) do not compile. But with the log output, I'm unable to tell why (maybe related to the cmake configuration ?).

**Comment by JonLiu1993** [comment]
@ctabin, sorry for the delayed reply, I will reproduce this error locally

**Comment by JonLiu1993** [comment]
@ctabin ,Is this PR being worked on?

**Comment by JonLiu1993** [comment]
@dg0yt ,Do you have any advice?

**Comment by ctabin** [comment]
@JonLiu1993 Unfortunately not. As said, I'm unable to provide more help here because I'm unable to see/understand the error (and also I'm not the author of `libzip`).

**Comment by dg0yt** [comment]
It is too late to look at ignition-fuel-tools4 CI build logs, and too much to do locally now. But it is clear that it needs proper exported configuration from libzip. So I did a build of libzip now. Inspecting the pc file indicates an issue wit zstd:
~~~
Libs:  -L"${libdir}" -lzip  -lbz2 -lZstd::Zstd -lz
~~~
Now looking for zstd in the exported cmake config indicates a missing `find_dependency`:
~~~
% grep -i zstd packages/libzip_x64-osx/share/libzip/libzip-*
packages/libzip_x64-osx/share/libzip/libzip-targets.cmake:  INTERFACE_LINK_LIBRARIES "\$<LINK_ONLY:BZip2::BZip2>;\$<LINK_ONLY:Zstd::Zstd>;\$<LINK_ONLY:ZLIB::ZLIB>"
~~~
This is just for a default build. Similar verification needs to be done for the optional features.

**Comment by JonLiu1993** [comment]
@dg0yt ,Thanks for your investigation and suggestions

**Comment by JonLiu1993** [comment]
@ctabin ，Have you verified according to @dg0yt's investigation and suggestions?

**Comment by JonLiu1993** [comment]
@ctabin ,Is this PR being worked on?

**Pull request closed without merging**